filters/filter.go

```diff
@@ -56,11 +56,32 @@ type Filter interface {
 }
 
 // NewFilter supports startsWith, endsWith, contains and a single wildcard
-// TODO(martinm): add rest of glob matching support and negation
+// along with negation
+// TODO(martinm): add rest of glob matching support
 func NewFilter(pattern string) (Filter, error) {
-	idx := strings.Index(pattern, WildcardChar)
+	nIdx := strings.Index(pattern, negationChar)
```

Comment:
Would it be more efficient if we scan the string once here and determine the indices of both negation char and wildchar instead of calling strings.Index multiple times?